diff --git a/dist/index.mjs b/dist/index.mjs
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -3,6 +3,25 @@
 
 //#region modules/analytics/client.ts
 const ANALYTICS_PORT = "@wxt-dev/analytics";
+function isBackgroundScript() {
+	if (!browser?.runtime?.getManifest || !location.protocol.endsWith("extension:")) return false;
+	const { background, background_page: backgroundPageWXT, manifest_version } = browser.runtime.getManifest();
+	const backgroundPageManifest = background && background.page;
+	const backgoundPageorScriptsCheck =
+		location.pathname === "/_generated_background_page.html" ||
+		(backgroundPageManifest && location.pathname.endsWith(backgroundPageManifest)) ||
+		(backgroundPageWXT && location.pathname.endsWith(backgroundPageWXT));
+	if (manifest_version === 3) {
+		try {
+			return self instanceof ServiceWorkerGlobalScope &&
+				background?.service_worker !== undefined &&
+				location.pathname.endsWith(background.service_worker);
+		} catch (e) {
+			return backgoundPageorScriptsCheck;
+		}
+	}
+	return backgoundPageorScriptsCheck;
+}
 const INTERACTIVE_TAGS = new Set([
 	"A",
 	"BUTTON",
@@ -21,7 +40,7 @@
 function createAnalytics(config) {
 	if (!browser?.runtime?.id) throw Error("Cannot use WXT analytics in contexts without access to the browser.runtime APIs");
 	if (config == null) console.warn("[@wxt-dev/analytics] Config not provided to createAnalytics. If you're using WXT, add the 'analytics' property to '<srcDir>/app.config.ts'.");
-	if (location.pathname === "/background.js") return createBackgroundAnalytics(config);
+	if (isBackgroundScript()) return createBackgroundAnalytics(config);
 	return createFrontendAnalytics();
 }
 /**
@@ -46,7 +65,8 @@
 		title: void 0
 	});
 	const getBaseEvent = async (meta) => {
-		const { arch, os } = await platformInfo;
+		const platform = (await platformInfo) ?? {};
+		const { arch, os } = platform;
 		return {
 			meta,
 			user: {
